File: src/org/usfirst/frc5293/util/preferences/BooleanPref.java
package org.usfirst.frc5293.util.preferences;

import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class BooleanPref extends Pref<Boolean> {
    public BooleanPref(String key, Boolean defaultValue) {
        super(key, defaultValue);
    }

    @Override
    public Boolean get() {
        return SmartDashboard.getBoolean(key, defaultValue);
    }

    @Override
    public void set(Boolean value) {
        SmartDashboard.putBoolean(key, value);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/framework/prefs/BooleanPref.kt
package org.usfirst.frc5293.framework.prefs

import edu.wpi.first.wpilibj.Preferences

class BooleanPref(key: String, default: Boolean) : Pref<Boolean>(key, default) {

    override fun get(): Boolean {
        return Preferences.getInstance().getBoolean(key, default)
    }

    override fun set(value: Boolean) {
        super.set(value)
        Preferences.getInstance().putBoolean(key, value)
    }
}
